src/lib.rs

```diff
@@ -1,5 +1,39 @@
+#![warn(rust_2021_compatibility, future_incompatible, nonstandard_style)]
+#![forbid(unsafe_code)]
+#![deny(bare_trait_objects, unused_doc_comments, unused_import_braces)]
+#![warn(missing_docs)]
+// Clippy:
+#![warn(clippy::all)]
+#![warn(clippy::pedantic)]
+#![warn(clippy::nursery)]
+#![warn(clippy::cargo)]
```

Comment:
I removed the rest since they were now default settings. Also removed `clippy.toml` since it made no difference with the settings you added. Unnecessary to have a file with settings that don't matter to us.